pkg/decoder.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 CloudWeGo Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hessian2
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+func NewHessian2Input(r io.Reader) *Decoder {
+	return &Decoder{
+		reader: r,
+		buffer: bytes.NewBuffer(nil),
+	}
+}
+
+type Decoder struct {
```

Comment:
no need to separate encoder/decoder.
pls refer to thriftCodec 
 https://github.com/cloudwego/kitex/blob/develop/pkg/remote/codec/thrift/thrift.go#L67